src/coreclr/System.Private.CoreLib/src/System/Runtime/DependentHandle.cs

```diff
@@ -71,7 +71,7 @@ public DependentHandle(object? target, object? dependent)
         /// <see cref="DependentHandle(object?, object?)"/> and has not yet been disposed.
         /// </summary>
         /// <remarks>This property is thread-safe.</remarks>
-        public bool IsAllocated => (nint)_handle != 0;
+        public readonly bool IsAllocated => (nint)_handle != 0;
```

Comment:
I understand that this is preventing a defensive copy at the IL level, but does this actually make any difference at all in the AOT or JIT?  I assume all the AOT compilers will optimize out extra copies since it is evident to them that `this` is not modified.  So that leaves RyuJIT. Perhaps it would be better to focus on additional analysis there to eliminate unnecessary copies.

Reply:
I wholeheartedly agree on some of the below cases, around setters, that it's likely confusing to mark them `readonly` even if they are only mutating indirect or static state.

But for cases like this, where its a trivial instance method that will never mutate. We should really just do the right thing and make everything simpler for Roslyn and the JIT.

Because even if the JIT is able to do all the right things, it has to do more to make it happen. That additional work can also sometimes block other optimizations due to other limits being hit around number of locals, total amount of IL inlined, etc. So it's ultimately better for us to mark cases like this correctly, which then also allows good interplay with other language features, analyzers, etc.

Reply:
> But for cases like this, where its a trivial instance method that will never mutate.

Agree, it is for clarity and a getter would seem to, in general, be a non-mutating operation.